MRML/vtkIGTLToMRMLPolyData.cxx

```diff
@@ -380,10 +380,9 @@ int vtkIGTLToMRMLPolyData::MRMLToIGTL(unsigned long event, vtkMRMLNode* mrmlNode
     
     //------------------------------------------------------------
     // Allocate Status Message Class
-    if (this->OutPolyDataMessage.IsNull())
-      {
-      this->OutPolyDataMessage = igtl::PolyDataMessage::New();
-      }
+    // Overwrite old message because we want to clear arrays from previous
+    // calls that will not be set anew.
+    this->OutPolyDataMessage = igtl::PolyDataMessage::New();
```

Comment:
@tokjun As @lassoan  mentioned, content from the previous message must be cleared before receiving a new one. I experienced a bug here when two polydata messages arrived but contained different structures. In this case the last message got content from the previous one.

If performance is an issue, one could manually clear each content of the structure instead, IF the new message don't contain such a structure. The code would be less elegant but maybe more efficient.